clients/guzzle6-adapter.rst

```diff
@@ -1,2 +1,52 @@
-Guzzle6 Adapter
-===============
+Guzzle 6 Adapter
+================
+
+An HTTPlug adapter for the `Guzzle 6 HTTP client`_.
+
+To install the Guzzle adapter, which will also install Guzzle itself (if it was
+not yet included in your project), run:
+
+.. code-block:: bash
+
+    $ composer require php-http/guzzle6-adapter
+
+Then begin by creating a Guzzle client, passing any configuration parameters you
+like::
+
+    $config = [
+        // Config params
+    ];
+    $guzzle = new GuzzleHttp\Client($config);
+
+Then create the adapter::
+
+    use Http\Adapter\Guzzle6\Client;
+
+    $adapter = new Client($guzzle);
+
+And use it to send synchronous requests::
+
+    use GuzzleHttp\Psr7\Request;
+
+    $request = new Request('GET', 'http://httpbin.org');
+
+    // Returns a Psr\Http\Message\ResponseInterface
+    $response = $adapter->sendRequest($request);
+
+Or send asynchronous ones::
+
+    use GuzzleHttp\Psr7\Request;
+
+    $request = new Request('GET', 'http://httpbin.org');
+
+    // Returns a Http\Promise\Promise
+    $promise = $adapter->sendAsyncRequest(request);
+
+Further reading
+---------------
+
+* Read more about :doc:`promises </components/promise>`.
+* Learn how you can decouple your code from any PSR-7 implementation (such as
+  Guzzle’s above) by using a :ref:`message factory <message-factory>`.
+
+.. _Guzzle 6 HTTP client: http://docs.guzzlephp.org/
```

Comment:
:+1: 
we might want to move this into a partial that we include right away, as this page will be used as template for other adapter / client documentations.